.github/workflows/spelling.yml

```diff
@@ -0,0 +1,25 @@
+name: Spell checking
+on:
+  pull_request_target:
```

Comment:
Could we do it so that PR branches are automatically fixed, and non-PR (read: `push`-triggered) builds will only warn?

Reply:
Can you clarify?

I suppose w/ 0.0.19 (again, I'm not quite done releasing 0.0.18), I could set it up so that when people make a PR, the bot automatically adds a commit with updates to `expect.txt`.

I've started the work to make something like this possible. And it's definitely an interesting workflow. If you're open to experimenting with it, then, yes, I'd be interested in trying this out.

The behavior would be something like:

1. User makes a fork (if they don't have write access to a repository)
2. User makes a branch
3. User adds a commit (including some foreign terms and a typo)
4. User pushes to a repository (which may or may not have actions / this action enabled)
5. User ignores the comment if it happens
6. User creates a PR to this repository
7. Bot automatically adds commit to update the expect file
8. If user doesn't notice the extra commit, when they try to make some other change to their branch, they'll get an error about being out of date -- I expect this to be somewhat confusing for people
9. User sees the note about what it did and then gets to decide if they want to replace the commit (which may involve a force push, or a pull, rewrite, and push-with-lease)
10. Reviewer is responsible for making sure that the expect changes aren't just rubber stamping typos -- this is technically always true whether or not an action like this exists

Note: the feature to write commits back into PRs is way too new / experimental for me to be ready to use it like this today (and I haven't written the extra code for the bot to be able to do it outside of a comment handler). I really want more implementation feedback before I try something this adventurous.

Reply:
Oh, sorry, I misunderstood... for some reason, I assumed that your plan was to fix spelling mistakes automatically, which I would have loved to do in PR builds (where I would even have gone so far as to allow the Action to add a commit and push it back to the contributor's branch), but not in CI builds.

Reply:
I think I've only seen one tool that can do that ([codespell](https://github.com/codespell-project/codespell)). And based on how I've seen it work, it seems to be pretty flaky/finicky. It has a fairly conservative list of corrections, a number of ambiguities, and a notable false-positive correction rate. (From my perspective, the biggest problem is it's too conservative, meaning that it doesn't flag a significant number of errors. I often run into it in that I'll run my tool on a project that has already been visited by that tool, and I'll still find & fix a significant number of errors.) -- It apparently does have an action flavor -- [actions-codespell](https://github.com/codespell-project/actions-codespell) -- this isn't an endorsement (I will probably look at its UX eventually), I've never seen the Action in use (and only found it just now, whereas I've seen the tool on a number of occasions).

The best I can imagine doing is the way old fashioned spell checkers worked where they offer suggestions at each site and a way to let people pick a replacement. -- And yes, that could involve pushing back into the repository. -- Actually, these days with GitHub's _suggestion_ feature, I suppose a bot could just make a whole bunch of inline suggestions and let the user try to collect them all. Offhand, I imagine that would be error-prone (if you miss one for a variable rename, you'd be very unhappy).

This isn't a feature I have today (or tomorrow in 0.0.18), but it's something I do imagine in the long run being possible. Current work has necessary steps along the road to doing it, but only the first steps amongst many.